macros/get_metafields.sql

```diff
@@ -1,6 +1,38 @@
-{% macro get_metafields(source_object, reference_value, lookup_object="stg_shopify__metafield", key_field="metafield_reference", key_value="value", reference_field="owner_resource") %}
+{% macro get_metafields(source_object, reference_values=None, lookup_object="stg_shopify__metafield", key_field="metafield_reference", key_value="value", reference_field="owner_resource") %}
+    {{ return(adapter.dispatch('get_metafields', 'shopify')(
+        source_object=source_object,
+        reference_values=reference_values,
+        lookup_object=lookup_object,
+        key_field=key_field,
+        key_value=key_value,
+        reference_field=reference_field
+    )) }} 
+{%- endmacro %}
 
-{% set pivot_fields = dbt_utils.get_column_values(table=ref(lookup_object), column=key_field, where="lower(" ~ reference_field ~ ") = lower('" ~ reference_value ~ "')") %}
+{% macro default__get_metafields(source_object, reference_values, lookup_object="stg_shopify__metafield", key_field="metafield_reference", key_value="value", reference_field="owner_resource") %}
+
+{# Ensure reference_values is defined and always a list #}
+{% if reference_values is none %}
+    {% do exceptions.raise_compiler_error("The reference_values parameter must be provided.") %}
+{% endif %}
```

Comment:
Is this necessary if we're going to be the ones exclusively using this macro in the Shopify package? It seems unnecessary and the `exceptions.raise_compiler_error` function is quite severe. If we don't meet that then the entire compilation of the project fails and all models are unable to run.

Since we're the only ones using this, I don't imagine this is entirely necessary. Removing the default value for `reference_values` in the macro call will essentially do the same. We can just ensure that we always have a value for this field.

For context, why don't we do the same for the `source_object` parameter?

Reply:
Agreed,  I was running into compilation issues early in development and added that in here to figure out why. But no longer necessary. Removed.